### PR TITLE
FEA Ranked probability score

### LINKED ISSUE #15: FEA Create Ranked Probability Score
Build an RPS metric. Very useful in the context of dataset shift with several timestamps (a model that with X tries to predict the timestamp that is and has good performance allows us to identify dataset shift).

image

### PR DESCRIPTION
**Pull request by jeancmaia**
Hey, peps! RPS is finally ready for the first round of reviews. Please, do not hesitate to leave our thoughts and feedback. Thank you :) 

fixes #15

### CODE REVIEW

**Review by vitaliset — COMMENTED** [review]
Here is a first pass of review for this feature. I still have to look at the "code itself" of the function but it looks good already. :)

**Review by vitaliset — APPROVED** [review]
Once the proposed modifications to `_transform_label_vector_to_matrix` are implemented, I will be happy to have this feature merged. It already looks great! Thanks for the awesome work, @jeancmaia. :D

**Review by dihanster — COMMENTED** [review]
Great job as always, Jean.
I agree with Carlo's points and just left a few more comments about repository defaults, but that's great!

File: datalib/metrics/_ranking.py [hunk 1]
```diff
@@ -142,3 +144,129 @@ def delinquency_curve(y_true, y_score, pos_label=None):
     optimal_rate = np.append(0, y_true_sorted.cumsum() / list_index)
 
     return approval_rate, default_rate, optimal_rate
+
+
+def ranked_probability_score_loss(
+    y_true, y_score, *, labels=None, sample_weight=None
```
[the remark this answers is not in the document]
Comment (dihanster) [reply datalib/metrics/_ranking.py:150]: ok

**Pull request closed without merging**